[jweeyh] iP

jweeyh opened the pull request.

Commit:
Add Gradle support

jweeyh committed:
Level 1

jweeyh committed:
created list function

jweeyh committed:
Added markings

jweeyh committed:
Created todo, deadline, event

jweeyh committed:
Added exceptions and delete

jweeyh committed:
Add save function to local data

jweeyh committed:
Change date from String type to LocalDate type

jweeyh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/DukeList.java

jweeyh committed:
Create more classes to abstract out common objects

jweeyh committed:
Merge branch 'add-gradle-support'

jweeyh committed:
Add Gradle

jweeyh committed:
Add JUnit tests for Parser and Task

jweeyh committed:
Create Jar file

jweeyh committed:
Create find function that checks if stored tasks have a given input substring

jweeyh committed:
Add JavaDocs documentation for commands, tasks and parser

jweeyh committed:
Check that code complies to coding standard

jweeyh committed:
Merge branch 'branch-A-CodingStandard'

jweeyh committed:
Check that code complies to coding standard

dfordarius commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,36 @@
+import exception.*;
+
+public class Duke {
+    private Storage storage;
+    private DukeList dukeList;
+    private Ui ui;
```
Perhaps these 3 variables can be final?

dfordarius commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,94 @@
+        } else {
+            String[] arr = s.split(" /from | /to", 3);
+            task = new Event(arr[0], arr[1], arr[2]);
+        }
+        System.out.println("Sure, Imma add that real quick");
+        System.out.println(task);
+        list.add(task);
+        System.out.println("Now you've got " + list.size() + pluralTask(list.size()));
+        System.out.println(new TextBorder(""));
+    }
```
Maybe you can move all the prints into the Ui class? After all you have a Ui class and the Ui class should deal with interactions with the user

dfordarius commented on src/main/java/command/MarkCommand.java:
```diff
@@ -0,0 +1,38 @@
+public class MarkCommand extends Command{
+
+    private String task;
+    private boolean marker;
```
Maybe change it to a more boolean sounding name like "isMarked" ?

bryansendeavour replied:
Maybe isMarked could be used for your boolean variable?

dfordarius commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+        String[] array = inputText.split(" ", 2);
+        try {
+            String first = array[0];
+            if (inputText.equals("bye")) {
+                return new ExitCommand(this.dukelist, this.storage);
+            } else if (first.equals("list")){
+                return new ListCommand(this.dukelist);
+            } else if (first.equals("mark")) {
+                return new MarkCommand(array[1], true, dukelist);
+            } else if (first.equals("unmark")) {
+                return new MarkCommand(array[1], false, dukelist);
+            } else if (first.equals("delete")){
+                return new DeleteCommand(array[1], dukelist);
+            } else if (first.equals("find")) {
+                return new FindCommand(array[1], dukelist);
+            } else {
+                if (array.length != 2) {
+                    if (first.equals("todo") || first.equals("deadline") || first.equals("event")) {
```
Maybe u can consider using switch for this part so it looks cleaner and improve readability?

dfordarius reviewed (commented):
Overall code quality is pretty good! Good OOP with some mixor fixes to be done! Keep up the good work 😄

dfordarius commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,94 @@
+        }
+    }
+
+    public void add(String type, String s) {
+        Task task = new Task();
+        if (type.equals("todo")) {
+            task = new Todo(s);
+
```
Maybe this method could be named to be more descriptive like "addTask"

jweeyh commented:
Hey Darius, thanks so much for the review!

On Tue, Jan 31, 2023 at 1:00 AM Darius Ng ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/duke/DukeList.java
> <https://github.com/nus-cs2103-AY2223S2/ip/pull/162#discussion_r1090899375>
> :
>
> > +    public void add(String type, String s) {
> +        Task task = new Task();
> +        if (type.equals("todo")) {
> +            task = new Todo(s);
> +
>
> Maybe this method could be named to be more descriptive like "addTask"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nus-cs2103-AY2223S2/ip/pull/162#pullrequestreview-1275576235>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU4UEONHH52UFRYH6G4NC6TWU7XRXANCNFSM6AAAAAAUEUCHMY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jweeyh committed:
Update README.md

bryansendeavour commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import task.*;
+import exception.*;
```
Would it be better to import all the classes instead?

bryansendeavour commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,94 @@
+
+    public DukeList() {}
+
+    private String pluralTask (int size) {
+        if (size == 1) {
+            return " task.";
+        } else {
+            return " tasks.";
+        }
+    }
```
Would it be better to put this in your Ui class?

bryansendeavour commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import command.*;
```
Maybe to conform to conventions you would like to import each class separately?

bryansendeavour commented on src/main/java/duke/TextBorder.java:
```diff
@@ -0,0 +1,22 @@
+
+public class TextBorder {
+    String text;
+    String border = "//  /  /  //  /  /  //  /  /  //  /  /  //  /  /  //  /  /  //  /";
```
Maybe this can be static final?

bryansendeavour reviewed (commented):
Code looks good, some minor suggestions made feel free to disregard if found unnecessary. Cheers!

jweeyh committed:
Add Gui

jweeyh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/command/ExitCommand.java
#	src/main/java/command/FindCommand.java
#	src/main/java/command/ListCommand.java
#	src/main/java/command/MarkCommand.java
#	src/main/java/duke/DukeList.java

jweeyh committed:
Merge remote-tracking branch 'origin/master'

jweeyh committed:
Add Gui

jweeyh committed:
Add Assert to SaveList in Storage

jweeyh committed:
Merge pull request #1 from jweeyh/branch-A-Assertions

Add Assert to SaveList in Storage

DerenC commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,57 @@
+
+    }
+
+    public DukeList retrieveList (Ui ui) {
+        try {
+            if ((new File(path)).exists()) {
+                FileInputStream fis = new FileInputStream(path);
+                ObjectInputStream ois = new ObjectInputStream(fis);
+                try {
+                    DukeList savedDukeList = (DukeList) ois.readObject();
+                    savedDukeList.setUi(ui);
+                    return savedDukeList;
+
+                } catch (IOException e) {
+                    ui.addStatement("Creating new save");
+                }
+            }
+
+        } catch (FileNotFoundException e) {
+            ui.addStatement("Hey sorry man, something's off bout you're save data.");
+            ui.addStatement(e.toString());
+        } catch (IOException e) {
+            ui.addStatement("Yo something's up with your I/O, gotta get it checked before doing this.");
+            ui.addStatement(e.toString());
+            System.out.println(e);
+        } catch (ClassNotFoundException e) {
+            ui.addStatement("Hey I can't find your object class.");
+            ui.addStatement(e.toString());
+        }
+        return new DukeList(ui);
+    }
```
Maybe can make this double nested try-catch block into a single level try-catch

DerenC commented on src/main/java/gui/Gui.java:
```diff
@@ -0,0 +1,118 @@
+        primaryStage.setResizable(false);
+        primaryStage.setMinHeight(600.0);
+        primaryStage.setMinWidth(400.0);
+
+        mainLayout.setPrefSize(400.0, 600.0);
+
+        scrollPane.setPrefSize(385, 535);
+        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
+        scrollPane.setVbarPolicy(ScrollPane.ScrollBarPolicy.ALWAYS);
+
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+
+        userInput.setPrefWidth(325.0);
+
+        sendButton.setPrefWidth(55.0);
+
+        AnchorPane.setTopAnchor(scrollPane, 1.0);
+
+        AnchorPane.setBottomAnchor(sendButton, 1.0);
+        AnchorPane.setRightAnchor(sendButton, 1.0);
+
+        AnchorPane.setLeftAnchor(userInput , 1.0);
+        AnchorPane.setBottomAnchor(userInput, 1.0);
+
+        sendButton.setOnMouseClicked((event) -> {
+            handleUserInput();
+        });
+
+        userInput.setOnAction((event) -> {
+            handleUserInput();
+        });
+
+        dialogContainer.getChildren().addAll(
+                new DialogBox(new Label("Waddup the name's Duncan. Sorry but Duke couldn't make it, had a pretty bad stomach-ache."), new ImageView(duncan)),
+                new DialogBox(new Label("So what do you need bro?"), new ImageView(duncan))
+        );
+    }
```
While the use of SLAP is quite good here, the method seems a bit too long

DerenC commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,68 @@
+                return new DeleteCommand(array[1], dukelist);
+            } else if (first.equals("find")) {
+                return new FindCommand(array[1], dukelist);
+            } else {
+                if (array.length != 2) {
+                    if (first.equals("todo") || first.equals("deadline") || first.equals("event")) {
+                        throw new EmptyDescriptionException("Sorry, your " + first + " task description is missing.");
+
+                    } else {
+                        throw new UnknownInputException("Hmm, I'm not sure what you're saying man.");
+
+                    }
+                }
+                return new AddCommand(array[0], array[1], dukelist);
+
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
```
This part of the code seems a bit too deeply nested

DerenC reviewed (commented):
A brief review of code quality, mainly on these problems
a. weak SLAP
b. arrow-head style code
c. too-long methods
d. too-deep nesting

jweeyh committed:
Change Parser to fit the coding standards

jweeyh committed:
Change Ui to fit the coding standards

There was text present that contained the default String drawing of Duke, which was relevant only in CLI, but has since become redundant with the new GUI.

This currently unused text is thus deleted in accordance with the coding standard to remove unused and outdated code.

Let's try to find other instances in the code which could be simplified and refactored.

jweeyh committed:
Add assert to the run function in Duke

At the start of the run function, it is important that the statements in the Ui are cleared so that previous commands would not be executed unintentionally.

An assert is thus placed after ui.clearStatements

This is done to confirm that the Ui statements have been cleared

jweeyh committed:
Merge pull request #4 from jweeyh/branch-A-Assertions

Add assert to the run function in Duke

jweeyh committed:
Merge branch 'master' into branch-A-CodingStandard

# Conflicts:
#	src/main/java/duke/Parser.java

jweeyh committed:
Merge pull request #3 from jweeyh/branch-A-CodingStandard

Branch Coding Standards

jweeyh committed:
Added archive and retrieve commands

As a matter of housekeeping and storage, there might be tasks that need to be removed from the current list, but may be retrieved in the future

An archive function that creates and stores another DukeList is implemented

As an extension, the archive function can help to store redundant tasks in the main list into a separate list for future retrieval

jweeyh committed:
Build shadowJar of project

jweeyh committed:
Add Ui.png to docs

jweeyh committed:
Update README.md

jweeyh committed:
Rename DukeException to DuncanException

jweeyh committed:
Merge remote-tracking branch 'origin/master'

jweeyh committed:
Update README.md

jweeyh committed:
Add error message when input is too short
